cargo-miri/src/setup.rs

```diff
@@ -137,32 +137,51 @@ pub fn setup(
     // not apply `RUSTFLAGS` to the sysroot either.
     let rustflags = &["-Cdebug-assertions=off", "-Coverflow-checks=on"];
 
-    // Do the build.
-    if print_sysroot || quiet {
-        // Be silent.
-    } else {
-        let mut msg = String::new();
-        write!(msg, "Preparing a sysroot for Miri (target: {target})").unwrap();
-        if verbose > 0 {
-            write!(msg, " in {}", sysroot_dir.display()).unwrap();
-        }
-        write!(msg, "...").unwrap();
-        if only_setup {
+    let mut msg = String::new();
```

Comment:
This is only needed inside the closure, it seems? Does it have to be computed outside?

Reply:
Constructing the message inside just requires giving the hook a lifetime that's not `'static` inside `rustc-build-sysroot.